powersimdata/network/zenodo.py

```diff
@@ -0,0 +1,144 @@
+import hashlib
+import json
+import os
+import shutil
+from contextlib import contextmanager
+from zipfile import ZipFile
+
+import requests
+from tqdm import tqdm
+
+url = "https://zenodo.org/api/records/"
+
+
+class Zenodo:
+    """Get data from a Zenodo archive.
+
+    :param str record_id: zenodo record id
+    """
+
+    def __init__(self, record_id):
+        """Constructor"""
+        self.record_id = record_id
+        self.content = self._get_record_content()
+
+    def _get_record_content(self):
+        """Make HTTP request to zenodo API and retrieve content.
+
+        :return: (*dict*) -- content of the response in unicode.
+        :raises Exception: if connection times out.
+        :raises ValueError: if record is invalid.
+        """
+        try:
+            r = requests.get(url + self.record_id, timeout=10)
+        except requests.exceptions.ConnectTimeout:
+            raise Exception("Connection to zenodo.org timed out")
```

Comment:
We could raise `ConnectionError` which is built-in

Reply:
Done